src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/service/LocationService.kt

```diff
@@ -231,6 +231,35 @@ class LocationService(
     )
   }
 
+  @Transactional
+  fun updateResidentialLocationUsedForTypes(id: UUID, usedFor: Set<UsedForType>): Boolean {
+    val residentialLocation = residentialLocationRepository.findById(id)
+      .orElseThrow { LocationNotFoundException(id.toString()) }
+
+    if (usedFor.isNotEmpty()) {
+      residentialLocation.updateCellUsedFor(
+        usedFor,
+        authenticationFacade.getUserOrSystemInContext(),
+        clock,
+      )
+    } else {
+      return false
+    }
+
+    log.info("Updated Used for type for Residential Location Children [$residentialLocation.getKey()]")
```

Comment:
`log.info("Updated used for types for cell locations under [$residentialLocation.getKey()]")`